lib/modules/manager/gleam/range.ts

```diff
@@ -0,0 +1,26 @@
+import { parseRange } from 'semver-utils';
+import { logger } from '../../../logger';
+import type { RangeStrategy } from '../../../types';
+import type { RangeConfig } from '../types';
+
+export function getRangeStrategy(config: RangeConfig): RangeStrategy {
+  const { currentValue, rangeStrategy } = config;
+  const isComplexRange = currentValue
+    ? parseRange(currentValue).length > 1
+    : false;
+
+  if (rangeStrategy === 'bump' && isComplexRange) {
+    logger.debug(
+      { currentValue },
+      'Replacing bump strategy for complex range with widen',
+    );
+    return 'widen';
+  }
+  if (rangeStrategy !== 'auto') {
+    return rangeStrategy;
+  }
+  if (isComplexRange) {
+    return 'widen';
+  }
+  return 'update-lockfile';
```

Comment:
This range strategy is not possible unless you parse the lock file and include `lockedVersion` in the extract phase